src/db/device/index.js

```diff
@@ -38,8 +38,23 @@ const createDevice = async (data) => {
   return new Model(data).save();
 };
 
+/**
+ * @description Updates device data in the database
+ * @param {object} deviceMatch An object describing how to select the device to be udpated
+ * @param {string} [deviceMatch._id] The unique id of the device
+ * @param {string} [deviceMatch.hospitalId] The hospitalId of the device
+ * @param {string} [deviceMatch.wardId] The wardId of the device
+ * @param {object} update The data to patch with the existing device data
+ * @returns {Promise} A promise that resolves or reject to the result of the database operation
+ */
+const updateDevice = async (deviceMatch, update) => {
+  const Model = deviceModel;
+  return Model.findOneAndUpdate(deviceMatch, { $set: update }, { new: true });
```

Comment:
I'm not sure we need the `$set` here, under the hood, Mongoose is going to add it before sending the query. [Source](https://mongoosejs.com/docs/api.html#model_Model.findOneAndUpdate).

Reply:
fixed